### DISCUSSION TITLE
Difference between UNet and RegUNet?

### DISCUSSION THREAD

**Discussion by ebrahimebrahim** [Project-MONAI/MONAI#2970] [Q&A] [answered]
What are some of the differences between [UNet](https://docs.monai.io/en/latest/networks.html?highlight=unet#unet) and [RegUNet](https://docs.monai.io/en/latest/networks.html?highlight=unet#regunet)?

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help share some comments here?
I think you know most of the details.

Thanks in advance.

**Comment by ericspod** [comment, marked as answer]
RegUNet is for image registration and has layers best suited for that task. Image registration is something we're actively working on in MONAI so there will be more on that topic in the future. The UNet class is meant to be a general purpose class for segmentation though it can be used for other purposes. It's layers can be changed by overridding its methods so it can serve as a template for more specialized versions. Unless you're doing registration this would be the one to use for now. We have other specalized versions of UNet that are meant to be referenc implementations of papers such as DynUNet or UNetR you can have a look at, being reference implementations they don't inherit the basic UNet class.

**Reply by ebrahimebrahim** [reply]
Thanks for explaining that!

I may be interested in contributing some registration-related work into MONAI in the future.
It would be nice to be aware of the current active work in that area.
I've already seen some of the contributions that came from the Deepreg project,
but please let me know if there's a way for me to learn about what is actively being worked on that hasn't yet made it into MONAI.
